clippy_lints/src/loops.rs

```diff
@@ -494,8 +494,40 @@ declare_clippy_lint! {
     "there is no reason to have a single element loop"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for iteration of `Option`s with
+    /// a single `if let Some()` expression inside.
```

Comment:
Hmm this is a little too specific since it does not include `Result`.

> Checks for iteration where the items are "manually flattened" such as using `if let Some(inner) = item { .. }`.

Reply:
You're right! I completely forgot to edit this part.